opentelemetry-sdk/src/metrics/internal/mod.rs

```diff
@@ -24,79 +23,65 @@ use crate::metrics::AttributeSet;
 pub(crate) static STREAM_OVERFLOW_ATTRIBUTES: Lazy<Vec<KeyValue>> =
     Lazy::new(|| vec![KeyValue::new("otel.metric.overflow", "true")]);
 
-/// Abstracts the update operation for a measurement.
-pub(crate) trait Operation {
-    fn update_tracker<T: Default, AT: AtomicTracker<T>>(tracker: &AT, value: T, index: usize);
-}
-
-struct Increment;
+pub(crate) trait Aggregator<T>
+where
+    T: Number,
+{
+    /// A static configuration that is needed in order to initialize aggregator.
+    /// E.g. bucket_size at creation time .
+    type InitConfig;
 
-impl Operation for Increment {
-    fn update_tracker<T: Default, AT: AtomicTracker<T>>(tracker: &AT, value: T, _: usize) {
-        tracker.add(value);
-    }
-}
+    /// Some aggregators can do some computations before updating aggregator.
+    /// This helps to reduce contention for aggregators because it makes
+    /// [`Aggregator::update`] as short as possible.
+    type PreComputedValue;
```

Comment:
nit: `PrecomputedValue` is not the best name as there is no precomputation happening for counters and gauges.

I think we could use something like `MeasurementData` instead.
```suggestion
    type MeasurementData;
```